File: snippets/csharp/VS_Snippets_ADO.NET/Classic WebData ConstraintCollection.AddRange Example/CS/source.cs
using System;
using System.Data;

public class Sample
{

    // <Snippet1>
    public static void ConstraintAddRange(DataSet dataSet)
    {
        try
        {
            // Reference the tables from the DataSet.
            DataTable customersTable = dataSet.Tables["Customers"];
            DataTable ordersTable = dataSet.Tables["Orders"];

            // Create unique and foreign key constraints.
            var uniqueConstraint = new UniqueConstraint(customersTable.Columns["CustomerID"]);
            var fkConstraint = new ForeignKeyConstraint("CustOrdersConstraint",
                customersTable.Columns["CustomerID"],
                ordersTable.Columns["CustomerID"]);

            // Add the constraints.
            customersTable.Constraints.AddRange(new Constraint[] 
                {uniqueConstraint, fkConstraint});
        }
        catch(Exception ex)
        {
            // Process exception and return.
            Console.WriteLine($"Exception of type {ex.GetType()} occurred.");
        }
    }
    // </Snippet1>

}

File: snippets/visualbasic/VS_Snippets_ADO.NET/Classic WebData ConstraintCollection.AddRange Example/VB/source.vb
Imports System.Data

Public Class Sample
    
    ' <Snippet1>
    Public Shared Sub ConstraintAddRange(dataSet As DataSet)
        Try
            ' Reference the tables from the DataSet.
            Dim customersTable As DataTable = dataSet.Tables("Customers")
            Dim ordersTable As DataTable = dataSet.Tables("Orders")

            ' Create unique and foreign key constraints.
            Dim uniqueConstraint As New _
                UniqueConstraint(customersTable.Columns("CustomerID"))
            Dim fkConstraint As New _
                ForeignKeyConstraint("CustOrdersConstraint", _
                customersTable.Columns("CustomerID"), _
                ordersTable.Columns("CustomerID"))

            ' Add the constraints.
            customersTable.Constraints.AddRange(New Constraint() _
                {uniqueConstraint, fkConstraint})

        Catch ex As Exception
            ' Process exception and return.
            Console.WriteLine($"Exception of type {ex.GetType().ToString()} occurred.")
        End Try
    End Sub
    ' </Snippet1>
End Class
